docs/apache-airflow/concepts/scheduler.rst

```diff
@@ -138,18 +141,172 @@ The following databases are fully supported and provide an "optimal" experience:
+  the processing might take a lot of CPU. You can mitigate it by decreasing the
+  :ref:`config:scheduler__min_file_process_interval`, but this is one of the mentioned trade-offs,
+  result of this is that changes to such files will be picked up slower and you will see delays between
+  submitting the files and getting them available in Airflow UI and executed by Scheduler. Optimizing
+  the way how your DAGs are built, avoiding external data sources is your best approach to improve CPU
+  usage. If you have more CPUs available, you can increase number of processing threads
+  :ref:`config:scheduler__parsing_processes`, Also Airflow Scheduler scales almost linearly with
+  several instances, so you can also add more Schedulers if your Scheduler's performance is CPU-bound.
+* Airflow might use quite significant amount of memory when you try to get more performance out of it.
+  Often more performance is achieved in Airflow by increasing number of processes handling the load,
+  and each process requires whole interpreter of Python loaded, a lot of classes imported, temporary
+  in-memory storage. This can lead to memory pressure. You need to observe if your system is not using
+  more memory than it has - which results with using swap disk, which dramatically decreases performance.
+  Note that Airflow Scheduler in versions prior to ``2.1.4`` generated a lot of ``Page Cache`` memory
+  used by log files (when the log files were not removed). This was generally harmless, as the memory
+  is just cache and could be reclaimed at any time by the system, however in version ``2.1.4`` and
+  beyond, writing logs will not generate excessive ``Page Cache`` memory. Regardless - make sure when you look
+  at memory usage, pay attention to the kind of memory you are observing. Usually you should look at
+  ``working memory``(names might vary depending on your deployment) rather than ``total memory used``.
+
+What can you do, to improve Scheduler's performance
+"""""""""""""""""""""""""""""""""""""""""""""""""""
+
+When you know what your resource usage is, the improvements that you can consider might be:
+
+* improve the logic, efficiency of parsing and reduce complexity of your DAG Python code. It is parsed
+  continuously so optimizing that code might bring tremendous improvements, especially if you try
+  to reach out to some external databases etc. while parsing DAGs (this should be avoided at all cost).
+  The :doc:`/best-practices` document shows a few examples on how you can approach dynamic DAG parsing
+  without reaching out to external sources.
+* improve utilization of your resources. This is when you have a free capacity in your system that
+  seems underutilized (again CPU, memory I/O, networking are the prime candidates) - you can take
+  actions like increasing number of schedulers, parsing processes or decreasing intervals for more
+  frequent actions might bring improvements in performance at the expense of higher utilization of those.
```

Comment:
Do we offer some way of identifying these?
For example I use a test to check how much time it takes to load the dag and from that I deduce if an expensive call was made in operator init.

Reply:
Yeah. That's the https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html#testing-a-dag which I will refer to.